### PR TITLE
Fix explorer links

### PR DESCRIPTION
**Pull request by austinhinderer**
**What current issue(s) from Trello/Github does this address?**
N/A

**What problem does this PR solve?**
The URL format for antchain explorer has changed, which means links to transactions are broken in the wallet.

**How did you solve this problem?**
Adjusted the URL format, and reworked the code for handling URLs to be more generic. I will get explorer.js to 100% coverage in another PR.

**How did you make sure your solution works?**
Tested that updated links worked now.

**Are there any special changes in the code that we should be aware of?**
There was a general refactor of how the explorer links are generated and how the data is stored.

**Is there anything else we should know?**
N/A

- [x] Unit tests written?

### CODE REVIEW

File: app/core/explorer.js [hunk 1]
```diff
@@ -1,77 +1,41 @@
...
+import explorerConfig from './explorerConfig'
+import { isMainNetwork } from './networks'
+
+export const buildExplorerLink = (
+  networkId: string,
+  explorer: ExplorerType,
+  structure: string,
+  suffix: string
+) => {
+  if (Object.prototype.hasOwnProperty.call(explorerConfig, explorer)) {
```
Comment (evgenyboxer) [inline app/core/explorer.js]: Why is this needed? won't `if (!!explorerConfig[explorer]) { ... }` suffice?

File: app/core/explorer.js [hunk 2]
```diff
@@ -1,77 +1,41 @@
...
-  const baseURL = getExplorerBaseURL(networkId, explorer)
-
-  if (explorer === EXPLORERS.NEO_TRACKER) {
-    return `${baseURL}/tx/${txId}`
-  } else if (explorer === EXPLORERS.NEO_SCAN) {
-    return `${baseURL}/transaction/${txId}`
-  } else {
-    return `${baseURL}/tx/hash/${txId}`
-  }
+  return ''
```
Comment (evgenyboxer) [inline app/core/explorer.js]: better throw an error... we want to know about this

File: app/core/explorerConfig.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    mainNetwork: 'https://neoscan.io/',
+    testNetwork: 'https://neoscan-testnet.io/',
+    trxLinkStructure: 'transaction/'
+  },
+  [EXPLORERS.ANT_CHAIN]: {
+    addressLinkStructure: 'address/info/',
+    assetLinkStructure: 'asset/hash/',
+    mainNetwork: 'http://antcha.in/',
+    testNetwork: 'http://testnet.antcha.in/',
+    trxLinkStructure: 'tx/hash/0x'
```
Comment (evgenyboxer) [inline app/core/explorerConfig.js]: sweet!

File: app/core/explorer.js (hunk 1 above)
Comment (austinhinderer) [reply app/core/explorer.js]: My personal feeling on !! is that's it's an anti-pattern. They do a similar thing in the end but I feel that the obj.hasOwnProperty code is more clear. The more verbose version is required because this is React.

File: app/core/explorer.js (hunk 2 above)
Comment (austinhinderer) [reply app/core/explorer.js]: Is there an example of an error you can provide?

File: app/core/explorer.js (hunk 1 above)
Comment (evgenyboxer) [reply app/core/explorer.js]: How about `explorerConfig.hasOwnProperty(explorer)` or even `(explorer in explorerConfig)`?

File: app/core/explorer.js (hunk 2 above)
Comment (evgenyboxer) [reply app/core/explorer.js]: maybe something like this:

`throw new Error(`${explorer} is not a valid explorer type. Valid explorer types are: ${Object.keys(explorerConfig).join(', ')}`)`

File: app/core/explorer.js (hunk 1 above)
Comment (austinhinderer) [reply app/core/explorer.js]: `explorerConfig.hasOwnProperty(explorer)` is not React compatible, which is why I used the longer syntax.

`in` will also return true if key gets found somewhere in the prototype chain, whereas Object.hasOwnProperty (like the name already tells us), will only return true if key is available on that object directly (its "owns" the property). [Citation](https://stackoverflow.com/questions/13632999/if-key-in-object-or-ifobject-hasownpropertykey)

Comment (evgenyboxer) [reply app/core/explorer.js]: What do you mean its not React compatible?

If you are doing a serious runtime validation, then your check is not sufficient. What would it help to know if the key just exists there? It can still fail if it doesn't adhere to a schema.
I recommend to use a schema validation, such as https://github.com/hapijs/joi. You can also write a flow type for `explorerConfig` to describe the shape.

Comment (austinhinderer) [reply app/core/explorer.js]: I had issues using obj.hasOwnProperty directly in React 15. They must have changed something about how objects are built for 16, but that means obj.hasOwnProperty works now.

**Comment by evgenyboxer** [comment]
Waiting for QA @drptbl

**Comment by drptbl** [comment]
@evgenyboxer 

Neotracker on testnet doesn't work for any transaction because it's not maintained anymore.

Antshares on mainnet last block update was 8 days ago which makes all my transactions lead to 404 pages (last transaction shown in my transaction history is from 5 days ago), example:
https://neoscan.io/transaction/706f8d22aaa1acc7be8fe64b92e453b0e88de1d4be4d27dac653bf128ecfb111
http://antcha.in/tx/hash/0x706f8d22aaa1acc7be8fe64b92e453b0e88de1d4be4d27dac653bf128ecfb111

Summary:
1. Neotracker works fine for mainnet but not for testnet
2. Antshares works fine for testnet but not for mainnet

**Comment by austinhinderer** [comment]
None of those problems are the result of this change, however. There are no regressions caused by this code, so should that block the release?

**Comment by drptbl** [comment]
@austinhinderer It's not blocking the release in any way. This pull request is named "Fix explorer links", so maybe it would be nice to sort out other issues with explorer links which are not working. That's why I've mentioned about other issues which are still there, so we can decide what to do next - fix them in this PR or fix them later.

**Comment by evgenyboxer** [comment]
Lets merge it and open tickets for the issues @drptbl found.
